### ISSUE TITLE
Update to Jekyll 4.0

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/extension-workshop#315]
See https://jekyllrb.com/news/2019/08/20/jekyll-4-0-0-released/ for details.

**Comment by caitmuenster** [comment]
Oh cool! @muffinresearch, do you have a sense of how much effort it will take to update to Jekyll 4.0? Also -- do you have a sense of the update will break anything we currently have set up?

**Comment by muffinresearch** [comment]
We'll need to make sure all of the plugins are compatible first.

Here's the list:

* [ ] jekyll-assets - see https://github.com/envygeeks/jekyll-assets/issues/613
* [ ] jekyll-seo-tag - Looks ok the tests have been updated to run against jekyll 4.0.
* [ ] jekyll_pages_api - it's not clear, we'd need to try it. They aren't testing against Jekyll 4 yet.
* [ ] jekyll-target-blank - unclear - also this doesn't look to be maintained so there's a fork.

**Comment by muffinresearch** [comment]
After seeing some progress on the deps, I've tried getting a branch running on Jekyll 4.0.

The bad news is this time to build from a cold cache:

```
$ jekyll --version && jekyll serve --unpublished --incremental --config _config.yml,_config_local.yml
jekyll 4.0.0
Configuration file: /Users/scol/code/extension-workshop/_config.yml
Configuration file: /Users/scol/code/extension-workshop/_config_local.yml
            Source: /Users/scol/code/extension-workshop
       Destination: /Users/scol/code/extension-workshop/_site
 Incremental build: enabled
      Generating...
                    done in 808.495 seconds.
```

That's > 13 minutes.

Also the resulting site doesn't quite work correctly. It might be too early to say if there aren't other issue causing the build to be this slow, but this doesn't look good.

**Comment by muffinresearch** [comment]
Closing this in favour of https://github.com/mozilla/extension-workshop/issues/654

**Comment by ashmaroli** [comment]
Hello @muffinresearch 
I'm part of the Jekyll Core Team and reached here from https://hacks.mozilla.org/2020/10/to-eleventy-and-beyond/

Intrigued by what could be causing such an abysmal performance, I set out to investigate on fork of repository.
Turns out, the culprit is actually the plugins used in the site.
With due respect to the authors of the plugins, *the build time came down to around 8.5s by disabling the third-party plugins itself. While the build time could have been further improved by disabling the first-party `jekyll-seo-tag` plugin as well, I did not bother.

To summarize, may I humbly request an edit be made to the above blog post stating the following:..?

*While Jekyll isn't slow, some of the plugins available can degrade the performance significantly..*

**Comment by muffinresearch** [comment]
> Hello @muffinresearch
> I'm part of the Jekyll Core Team and reached here from https://hacks.mozilla.org/2020/10/to-eleventy-and-beyond/
> 
> Intrigued by what could be causing such an abysmal performance, I set out to investigate on fork of repository.
> Turns out, the culprit is actually the plugins used in the site.
> With due respect to the authors of the plugins, *the build time came down to around 8.5s by disabling the third-party plugins itself. While the build time could have been further improved by disabling the first-party `jekyll-seo-tag` plugin as well, I did not bother.
> 
> To summarize, may I humbly request an edit be made to the above blog post stating the following:..?
> 
> _While Jekyll isn't slow, some of the plugins available can degrade the performance significantly.._

@ashmaroli thanks for digging into this, I've added a clarifying update to the post.
